### DISCUSSION TITLE
Testing soc nv flash driver on STM32H747i-disco using nvs subsystem sample

### DISCUSSION THREAD

**Discussion by mkaricheri** [zephyrproject-rtos/zephyr#64880] [General]
Hi Experts,

I need to define multiple partitions (assume requires flash_map enabled) on the SoC Flash on STM32H747i-disco. STM32 has a flash controller driver and that is what I want to test first. So I tried building samples/subsys/fs/nvs and I get a flash init error because of invalid sector size. I have put debug prints and found driver is reporting sector size of 128KiB (131072) which I think is correct. By nvs code is complaining it sees 0 for sector size ([00:00:00.082,000] <err> fs_nvs: Invalid sector size 0). Am I missing something? I have made the changes to DTS here. https://github.com/zephyrproject-rtos/zephyr/commit/b732cb956d9d16229eade4e202ebfbfe8432c871

Could you tell me what is missing in my DTS?

This is supported on some of the STM boards namely nucleo_* here https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/subsys/nvs/boards

My end goal is to enable flash_map on SoC internal flash and have application write images or configuration data to the partition. Something similar to https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/soc_flash_nrf using flash API

So any help would be appreciated. 

Murali

**Comment by mkaricheri** [comment]
My debug logs are part of the commit message

**Reply by mkaricheri** [reply]
@erwango could you please help me out on this? I am looking for guidance as to what is supported and what is missing for this?

**Reply by erwango** [reply]
@FRASTM ^^

**Comment by Laczen** [comment]
@mkaricheri NVS does not support a sector size of 128kB. On top of this your dts definition would only have one sector and there would not be persistent storage.

**Reply by FRASTM** [reply]
Correct, `struct nvs_fs` has a sector_size value on 16bits and the FLASH_SECTOR_SIZE is defined by the ../modules/hal/stm32/stm32cube/stm32h7xx/soc/ to be 0x20000 on this stm32h747 serie (given by the flash_stm32_page_layout of the flash_stm32h7x driver).

I realize that such Error could be better reported by the LOG_ERR of the nvs driver with a following return msg 
```
diff --git a/samples/subsys/nvs/src/main.c b/samples/subsys/nvs/src/main.c
index 5ea1976d31..3c70477444 100644
--- a/samples/subsys/nvs/src/main.c
+++ b/samples/subsys/nvs/src/main.c
@@ -89,6 +89,13 @@ int main(void)
 		printk("Unable to get page info\n");
 		return 0;
 	}
+
+	if (info.size > UINT16_MAX) {
+		/* fs.sector_size is a 16bit value */
+		printk("ERROR: sector_size cannot exceed 0xFFFF\n");
+		return 0;
+	}
+
 	fs.sector_size = info.size;
 	fs.sector_count = 3U;
```

**Comment by FRASTM** [comment]
also reported by https://github.com/zephyrproject-rtos/zephyr/issues/19895
